docs/design/connection-tester.md

```diff
@@ -0,0 +1,50 @@
+
+## Side effect free connection test
+
+Endpoints like database or REST endpoints can be checked side-effect free (e.g. for HTTP based services on could try a `HEAD` request to the root context). If this is not possible because e.g. some payload would be needed to be transferred in order to determine whether the connection works, then a connection test in the sense described here can't be performed. Instead some "sample flow run" needs to be performed which moves a test payload through the flow where the connector is at the the end (but this should be also represented differently on the UI).
+
+Its important that the UI knows whether a connection is testable or not so that it can provide the appropriate UI state (e.g. an enable or disabled test button). Therefore each `Connector` has an extra `canPing()` method to allow to query for this information. `ipaas-rest` stores this information along with the `Connection` object which is handed out to the UI for showing the connection (along with the 'test' button). The method `canPing()` should receive the same `ConnectorConfiguration` as `ping()` itself so that it can dynamically determine whether a ping is possible. By default this argument to `canPing()` is probably ignored since a connector's singable-attribute is unique to the type of connector and independent of connection parameters.
+
+In summary, `ping()` and `canPing()` are suggested as additions to the connector interface within the upstream Camel connector API definitions.
+
+## Connection Tester Runtime
+
+It is essential, that connection testing happens as fast as possible as it is a synchronous operation. The initial trigger is a HTTP request from `ipaas-ui` to `ipaas-rest` which then performs the test action against the backend. A naive approach would be to spin up a Java pod with the referenced connector, call the `ping()` method, return the result to the client and (asynchronously) tears down the pod. However, the startup time are considered to be too expensive, to that it make sense to have a pool of pods running which can be directly contacted to perform the ping check. Various scenarios are conceivable:
+
+* **All-in-one** pods which can perform the ping check for all connectors. These would have access to all connectors supported and expose a service which has the connector type as parameter as well as the `ConnectorConfiguration` for calling out to the connector. A single K8s service is in front of these pods which is called by the `ipaas-rest`. This will work only if all connectors are known upfront or connectors could be loaded dynamically. 
```

Comment:
For the ipaas demo we could have this approach with a already warm pod with all the possible connectors.